crates/bevy_ui/src/focus.rs

```diff
@@ -111,16 +116,36 @@ pub fn ui_focus_system(
     let cursor_position = camera
         .iter()
         .filter(|(_, camera_ui)| !is_ui_disabled(*camera_ui))
-        .filter_map(|(camera, _)| {
-            if let RenderTarget::Window(window_id) = camera.target {
-                Some(window_id)
-            } else {
-                None
+        .find_map(|(camera, _)| {
+            match &camera.target {
+                RenderTarget::Window(window_id) => windows.get(*window_id).and_then(|window| {
+                    window
+                        .is_focused()
+                        .then(|| window.cursor_position())
+                        .flatten()
+                }),
+                RenderTarget::Image(handle) => images
```

Comment:
This doesn't really make sense to me. When rendering to an Image, there is no way to know if (or where) that image will be rendered relative to the cursor. In its current form, the behavior here is "undefined" and only works correctly in some cases.

To cover this scenario, I think what we really want is some sort of RenderTarget UI widget that receives the cursor information in window-space, then converts that to local widget space, then converts that to some widget-specific input event.

Reply:
This will be a relatively challenging design space, but I don't think going down the path laid out in this PR is worth it, despite it being much easier to implement.

Reply:
That's fair enough, I'll try to have a think of how this could be accomplished in a better way.

Reply:
I wonder if it would make sense to take the most flexible route, and be able to tell the UI what the cursor position is manually. Then it could work for all kinds of cases, like if the ui was mapped onto a sphere, for example.

Reply:
Another idea is add a field to `UiCameraConfig` called `interaction_size` which takes an enum:
```rust
enum InteractionSize {
    /// Interaction is based on window size. This is how Bevy currently works.
    Window,
    /// Interaction is scaled based on image texture size. This is what my PR initially added.
    Scaled,
    /// Interaction is scaled based on a custom fixed size.
    Sized(f32, f32),
}
```